lib/Genesis/Manifest.pm

```diff
@@ -0,0 +1,168 @@
+package Genesis::Manifest;
+
+use base 'Exporter';
+our @EXPORT = qw{
+  new
+};
+
+use constant {
+	REDACTED_FILENAME   => "redacted-manifest.yml",
+	UNREDACTED_FILENAME => "unredacted-manifest.yml"
+};
+
+use JSON::PP;
+use File::Path qw(rmtree);
+
+use Genesis::Run;
+use Genesis::Utils;
+
+sub new {
+	my ($class,$env,%opts) = @_;
+
+	my $self = {
+		env               => $env,													# Will be replaced with a Genesis::Env object when available.
+		type              => delete $opts{type} || main::deployment_suffix($env),	# Genesis::Env will have this in the future
+		cloud_config      => delete $opts{'cloud-config'},
+		create_env        => main::is_create_env($env),
+		workdir           => workdir("$env-manifest"),
+		files             => undef
+	};
+
+	# Special circumstance if a create-env to populate a special cloud-config
+	if ($self->{create_env}) {
+		$self->{cloud_config} = $self->{workdir}."/cloud.yml";
+		main::write_stemcell_data($self->{cloud_config});
+	}
+	return bless($self,$class);
+}
+
+sub DESTROY {
+	my ($self) = @_;
+	if ($self && $self->{workdir} && -d $self->{workdir}) {
+		rmtree $self->{workdir};
+		main::debug("#g{Deleted Manifest workdir:} $self->{workdir}");
+	}
+}
+
+# Returns the source files used to build the manifest.  Option no_base will
+# exclude the genesis-generated base file, and option no_kit will exclude the
+# yaml files that come from the kit
+
+sub source_files {
+	my ($self,%opts) = @_;
+	my @files = ();
+	push @files, $self->_base_yml_file unless $opts{no_base};
+	push @files, $self->_kit_yaml_files unless $opts{no_kit};						# Will be replaced with Genesis::Kit obj method call
+	push @files, main::mergeable_yaml_files($self->{env});							# Will be replaced with Genesis::Env obj method call
```

Comment:
Here you can see forensic evidence that the code author possessed too large of a monitor.